KabuKit/Classes/core/Swift/Screen.swift

```diff
@@ -1,72 +1,82 @@
 import Foundation
 
+fileprivate var suspendCallback: [ScreenHashWrapper : () -> Void] = [ScreenHashWrapper : () -> Void]()
+fileprivate var resumeCallback: [ScreenHashWrapper : () -> Void] = [ScreenHashWrapper : () -> Void]()
 
+fileprivate var screenBehaviorContainer: [ScreenHashWrapper: ScreenBehavior] = [ScreenHashWrapper: ScreenBehavior]()
+
+/**
+ 表示される画面のProtocolです
+ 
+ */
 public protocol Screen : class {
     
+    
+    var isSuspended: Bool { get }
+        
     /**
-     リンク先に遷移する
-     
-     - Attention:
-     指定リンク先がなにであるかはScenario側で指定しておく必要がある
+     画面がサスペンド状態になった際に呼ばれます
      
-     それを忘れるとこのメソッドを呼んでもなにも起きない  
-
-     
-     - Parameters:
-       - request: 遷移先へのリンク
      */
-    func send<ContextType>(_ request: Request<ContextType>, _ completion: @escaping (Bool) -> Void) -> Void
-    
-    
-    func send<ContextType>(_ request: Request<ContextType>) -> Void
+    func onSuspend() -> Void
     
     /**
-     現在表示されているSceneを終了させ、前のSceneに戻る
+     画面がアクティブ状態になった際に呼ばれます
      
-     - Attention :
-     ただし、前のシーンがない場合は戻らず、何も起きない
      */
-    func leave() -> Void
-
-    func leave(_ runTransition: Bool) -> Void
-
-    func leave(_ completion: @escaping (Bool) -> Void) -> Void
+    func onActivate() -> Void
+}
 
-    func leave(_ runTransition: Bool, _ completion: @escaping (Bool) -> Void) -> Void
+public extension Screen {
+    public var isSuspended: Bool {
+        guard let behavior = screenBehaviorContainer[ScreenHashWrapper(self)] else {
+            return true
```

Comment:
Is this return value is appropriate?

Reply:
Yes.
When `Screen` is generated, `Screen`'s status is `suspend`.
So, if `ScreenBehavior` is not set, `Screen`'s status must be `suspend`.

I'll write that's comment in japanese 🤒

Reply:
Thanks :)